src/components/PostsList.tsx

```diff
@@ -87,0 +1,120 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { Post } from '../types/Post';
+import classnames from 'classnames';
+import { getPostsByUserId } from '../api/posts';
+import { User } from '../types/User';
+import { Loader } from './Loader';
+
+type Props = {
+  selectedUser: User | null;
+  onPostSelect: (post: Post | null) => void;
+};
+
+export const PostsList: React.FC<Props> = ({ onPostSelect, selectedUser }) => {
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(false);
+
+  const handleOpenPost = useCallback(
+    (post: Post) => {
+      if (selectedPost === post) {
+        setSelectedPost(null);
+        onPostSelect(null);
+      } else {
+        setSelectedPost(post);
+        onPostSelect(post);
+      }
+    },
+    [selectedPost, onPostSelect],
+  );
+
+  useEffect(() => {
+    const loadPosts = async () => {
+      if (!selectedUser?.id) {
+        return;
+      }
+
+      setIsLoading(true);
+      setError(false);
+
+      try {
+        const postsFromServer = await getPostsByUserId(selectedUser.id);
+
+        setPosts(postsFromServer);
+      } catch {
+        setError(true);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    loadPosts();
+    setSelectedPost(null);
```

Comment:
You are using `setState` several times in one function call (method). This could lead to unnecessary re-renders and make the code harder to read. Consider grouping these calls together.